x/metadata/keeper/msg_server.go

```diff
@@ -98,7 +103,10 @@ func (k msgServer) AddScopeDataAccess(
 
 	existing.AddDataAccess(msg.DataAccess)
 
-	k.SetScope(ctx, existing)
+	err := k.SetScope(ctx, existing)
+	if err != nil {
+		return nil, fmt.Errorf("could not update scope %q: %w", msg.ScopeId, err)
+	}
```

Comment:
It's not being redeclared. There are other variables named `err` in the same function, but they're all scoped to the `if` statements that they're defined for. Trying to use `=` here instead of `:=` is a syntax/compilation error.